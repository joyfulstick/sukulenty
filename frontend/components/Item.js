import React, { Component } from 'react'
import DeleteItem from './DeleteItem'
import ItemStyles from './styles/ItemStyles'
import Link from 'next/link'
import PriceTag from './styles/PriceTag'
import PropTypes from 'prop-types'
import Title from './styles/Title'
import formatMoney from '../lib/formatMoney'

class Item extends Component {
  render() {
    const {
      item: { id, title, price, description, image },
    } = this.props
    return (
      <ItemStyles>
        {image && <img src={image} alt={title} />}
        <Title>
          <Link
            href={{
              pathname: '/item',
              query: { id },
            }}
          >
            <a>{title}</a>
          </Link>
        </Title>
        <PriceTag>{formatMoney(price)}</PriceTag>
        <p>{description}</p>
        <div className="buttonList">
          <Link
            href={{
              pathname: 'update',
              query: { id },
            }}
          >
            <a>Edytuj üìù</a>
          </Link>
          <button>üõí Dodaj do koszyka</button>
          <DeleteItem id={id} image={image}>‚ùé Usu≈Ñ przedmiot</DeleteItem>
        </div>
      </ItemStyles>
    )
  }
}

Item.propTypes = {
  item: PropTypes.object.isRequired,
}

export default Item
