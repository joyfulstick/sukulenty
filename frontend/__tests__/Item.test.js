import Item from '../components/Item'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

const fakeItem = {
  id: '1df5f2',
  title: 'Nazwa',
  price: 799,
  description: 'Opis',
  image: 'item.jpg',
  largeImage: 'lgitem.jpg',
}

describe('<Item />', () => {
  it('renders', () => {
    shallow(<Item item={fakeItem} page={1} />)
  })
  it('matches the snapshot', () => {
    const wrapper = shallow(<Item item={fakeItem} page={1} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
