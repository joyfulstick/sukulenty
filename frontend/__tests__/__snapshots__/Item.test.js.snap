// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> matches the snapshot 1`] = `
<ItemStyles__Item>
  <Link
    href={
      Object {
        "pathname": "/item",
        "query": Object {
          "id": "1df5f2",
        },
      }
    }
  >
    <a>
      <img
        alt="Nazwa"
        src="item.jpg"
      />
      <Title>
        <h3>
          Nazwa
        </h3>
      </Title>
      <PriceTag>
        7,99¬†z≈Ç
      </PriceTag>
      <p>
        Opis
      </p>
    </a>
  </Link>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "/update",
          "query": Object {
            "id": "1df5f2",
          },
        }
      }
    >
      <a>
        Edytuj üìù
      </a>
    </Link>
    <AddToCart
      id="1df5f2"
    />
    <DeleteItem
      id="1df5f2"
      image="item.jpg"
      page={1}
    >
      ‚ùé Usu≈Ñ przedmiot
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
